app/src/main/java/com/infomaniak/mail/ui/main/thread/actions/AttachmentActionsBottomSheetDialog.kt

```diff
@@ -71,45 +69,37 @@ class AttachmentActionsBottomSheetDialog : ActionsBottomSheetDialog() {
     }
 
     private fun setupListeners(attachment: Attachment) = with(binding) {
+
+        fun navigateToDownloadProgressDialog(attachmentIntentType: AttachmentIntentType) {
+            safeNavigate(
+                resId = R.id.downloadAttachmentProgressDialog,
+                args = attachment.createDownloadDialogNavArgs(attachmentIntentType),
+                currentClassName = AttachmentActionsBottomSheetDialog::class.java.name,
+            )
+        }
```

Comment:
It's not supposed to be used elsewhere than inside this function and we already have access to the attachment object if we make it an inner function so we don't need to make it more complicated by extracting it outside to then pass in the variable, we already have access to more easily, three times I think

Reply:
I find that inner functionare more useful  to compute value needed only in a specific function rather than willing to put all the different logics that could be in their own function.
Or at least when there is a lots of parameters needed, here it's just one, it's nothing

Reply:
Anyway, this function needs to be moved out in a later PR